From 2eecec86589b35d309e9ffd7e37efae0355fe41d Mon Sep 17 00:00:00 2001
From: "sangmo.kang" <sangmo.kang@lge.com>
Date: Tue, 12 Oct 2021 07:04:00 +0000
Subject: [PATCH] Fix compile error with g++ 7.5.0

---
 src/corelib/text/qregularexpression.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/corelib/text/qregularexpression.cpp b/src/corelib/text/qregularexpression.cpp
index e055ae064a..6d8dff0cfa 100644
--- a/src/corelib/text/qregularexpression.cpp
+++ b/src/corelib/text/qregularexpression.cpp
@@ -1179,13 +1179,14 @@ void QRegularExpressionPrivate::doMatch(QRegularExpressionMatchPrivate *priv,
     // subject may have data == nullptr. In this case, to keep PCRE
     // happy, pass a pointer to a dummy character.
     constexpr char16_t dummySubject = 0;
+    const char16_t * const pDummySubject = &dummySubject;
     const char16_t * const subjectUtf16 = [&]()
     {
         const auto subjectUtf16 = priv->subject.utf16();
         if (subjectUtf16)
             return subjectUtf16;
         Q_ASSERT(subjectLength == 0);
-        return &dummySubject;
+        return pDummySubject;
     }();
 
     int result;
-- 
2.32.0

